### ISSUE TITLE
Correct Specimen type and Device type's label in the test card form

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#8550]
## Area or Component
Conduct test page with a test card

## Quick Summary

The label for the "Test device" field and "Specimen type" field has the `for` attribute set to a number but the `<Select>`'s `id` attribute set to `test-device-<UUID>`. Because the `for` and `id` are not the same value this is unclear what the label for this input is. This is also why Deque is flagging as the label being missing.

## Accessibility Violation

[label-not-present](https://docs.deque.com/issue-help/1.0.0/en/label-not-present)

## Action Requested

Update the label of the "Device type" and "Specimen type" labels to include the correct `for` value.

## Screenshots (optional)

<img width="1812" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd2tUQ1E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1337487a86cd37de817a70160901240254b72d16/Screenshot%202025-02-28%20at%2014.25.50.png" alt="Screenshot 2025-02-28 at 14.25.50.png" />

## Source Code (optional)

https://github.com/CDCgov/prime-simplereport/blob/c862a94015ce46f67918b528ad3c89e7f7b4e75b/frontend/src/app/testQueue/TestCardForm/TestCardForm.tsx#L497

https://github.com/CDCgov/prime-simplereport/blob/c862a94015ce46f67918b528ad3c89e7f7b4e75b/frontend/src/app/testQueue/TestCardForm/TestCardForm.tsx#L527
